cms/djangoapps/contentstore/views/tabs.py

```diff
@@ -77,7 +75,33 @@ def tabs_handler(request, course_key_string):
         return HttpResponseNotFound()
 
 
-def reorder_tabs_handler(course_item, request):
+def get_course_tabs(course_item: CourseBlock, user: User) -> Iterable[CourseTab]:
+    """
+    Yields all the course tabs in a course including hidden tabs.
+    """
+
+    for tab in CourseTabList.iterate_displayable(course_item, user=user, inline_collections=False, include_hidden=True):
+        if isinstance(tab, StaticTab):
+            # static tab needs its locator information to render itself as an xmodule
+            static_tab_loc = course_item.id.make_usage_key('static_tab', tab.url_slug)
+            tab.locator = static_tab_loc
+        yield tab
+
+
+def update_tabs_handler(course_item, tabs_data, user):
+    """
+    Helper to handle updates to course tabs based on API data.
+    """
+
+    if 'tabs' in tabs_data:
+        reorder_tabs_handler(course_item, tabs_data, user)
+    elif 'tab_id_locator' in tabs_data:
+        edit_tab_handler(course_item, tabs_data, user)
+    else:
+        raise NotImplementedError('Creating or changing tab content is not supported.')
```

Comment:
Is this user-facing error? does this need i18n?

Reply:
I have not changed any existing code in this file, just refactored it. This code was just moved over here from above. 
This code is used by studio's django-driven frontend currently. Unless someone is messing with the request structure, this error won't be triggered. 

If the case is triggered for some reason in the current UI, it result in a 500 error since it is uncaught. 
That isn't ideal, but I have avoided changing anything the existing UI uses in order to avoid breaking anything code or tests that rely on existing behaviour.

Reply:
@xitij2000 - when you say you "have not changed the existing code", do you mean 
1. this code already existed?
2. if yes, can you point to that code?
3. What is the reason for this refactoring?

Reply:
1. Yes. 
2. Here it is: https://github.com/edx/edx-platform/pull/27608/files#diff-34bdd73f7fac8f6081b034a2dff267a288733803d3dd3a6f50c2816cf49bb1c3L51-L56
3. This refactoring is to enable better code reuse between the existing API and the new API. 

I think the ADR should give more context here, but to boil it down:

Currently, the only way to edit course advanced settings is via studio. The API that studio uses only supports session-based authentication, so that makes in inaccessible to MFEs. The course authoring MFE will need to change some advanced course settings, so we need an API for that. 

As a first/quick step in enabling that, this PR reuses the existing code, but wraps it in a new API so that MFEs can start using these APIs. I don't want to break any existing usage, so I'm keeping the changes to existing code minimal. This refactoring just splits out reusable chunks of existing code into a new functions so they can be called by the DRF views.

Reply:
thanks for putting this together, appreciate that. I have also read the ADR, and its makes much more sense now.